cluster-cm/src/main/java/com/sequenceiq/cloudbreak/cm/ClouderaManagerDeployClientConfigProvider.java

```diff
@@ -0,0 +1,253 @@
+                throw new CloudbreakException(
+                        String.format("%s request against CM API returned with status code: %d",
+                                CMF_COMMANDS_COMMAND_TABLE_URI_PATH, response.code()));
+            }
+            return getCommandIdFromCMResponse(response);
+        } catch (IOException e) {
+            throw new CloudbreakException(
+                    String.format("DeployClusterClientConfig - error during processing %s CM request",
+                            CMF_COMMANDS_COMMAND_TABLE_URI_PATH), e);
+        }
+    }
+
+    private BigDecimal getCommandIdFromCMResponse(Response response) throws IOException {
+        BigDecimal result = null;
+        try (ResponseBody responseBody = response.body()) {
+            String responseBodyStr = responseBody.string();
+            LOGGER.debug("Start processing commandTable response: {}", responseBodyStr);
+            List<CommandResource> commandList =
+                    new ObjectMapper().readValue(responseBodyStr, new TypeReference<>() {
+                    });
+            LOGGER.debug("Processing commandTable were successful.");
+            long currentStartTime = Long.MIN_VALUE;
+            for (CommandResource command : commandList) {
+                if (DEPLOY_CLIENT_CONFIG_COMMAND_NAME.equals(command.getName())) {
+                    LOGGER.debug("Found DeployClusterClientConfig command based on "
+                                    + "{} response [command_id: {}]", CMF_COMMANDS_COMMAND_TABLE_URI_PATH,
+                            command.getId());
+                    long startTime = command.getStart();
+                    if (startTime > currentStartTime) {
+                        LOGGER.debug("Found latest DeployClusterClientConfig command " +
+                                "with [command_id: {} by commandTable call]", command.getId());
+                        currentStartTime = startTime;
+                        result = new BigDecimal(command.getId());
+                    }
+                }
+            }
+        }
+        return result;
+    }
+
+    // similar as apiClient#Call creator methods - so keep this in one place
+    private Request createRequest(ApiClient apiClient, Map<String, List<String>> headers)
+            throws ApiException {
+        List<Pair> queryParams = new ArrayList<>();
+        Map<String, String> headerParams = new HashMap<>();
+        String[] localVarAccepts = new String[]{"application/json"};
+        String localVarAccept = apiClient.selectHeaderAccept(localVarAccepts);
+        if (localVarAccept != null) {
+            headerParams.put("Accept", localVarAccept);
+        }
+        String[] localVarContentTypes = new String[0];
+        String localVarContentType = apiClient.selectHeaderContentType(localVarContentTypes);
+        headerParams.put("Content-Type", localVarContentType);
```

Comment:
read the comment on the method, that's technically the same mechanism that is used in ApiClient. so im using the same initialization how it would be look like if ApiClient would have a valid api call for getting the recent commands